File: Algorithm/Programmers/위장.py
from collections import defaultdict
def solution(clothes):
    answer = 1
    d = defaultdict(list)
    
    for i in range(len(clothes)):
        d[clothes[i][1]].append(clothes[i][0])
    
    #print(d)
    
    for v in d.values():
        answer *= len(v) + 1
    
        
    return answer - 1

File: Algorithm/Programmers/위장.java
import java.util.*;
    
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> m = new HashMap<>();
        
        for (int i = 0; i < clothes.length; i++){
            String key = clothes[i][1];
            m.put(key, m.getOrDefault(key, 0) + 1);
        }
        
        for (int mValue : m.values()){
            //System.out.println(mValue);
            answer *= mValue + 1;
        }
        
        return answer - 1;
    }
}